cookbook/brilliant-mirabella-genio-smart-plugs.rst

```diff
@@ -0,0 +1,258 @@
+#. If you haven't already connect your smart plug and turn it on. It should automatically enter pairing mode, if it doesn't follow the instructions below:
+
+   * Brilliant `(PDF Manual) <https://www.brilliantsmart.com.au/wp-content/uploads/2019/02/03.-How-to-setup-the-Smart-plug.pdf>`__
+
+    * To connect the Smart WiFi Plug to your WiFi press and hold the manual control button for about 5 seconds until you hear a click. Let go of the button
+      as soon as you hear the **click**. Then press button once again to activate blinking. **The red indicator light should now blink rapidly (2 times per
+      second)**.
+
+   * Mirabella `(PDF Manual) <https://img1.wsimg.com/blobby/go/67cdd3b9-3600-4104-b097-603c05201237/downloads/1cq5h6nt7_336441.pdf>`__
+
+    * To do this, ensure your plug is on (the indicator light should be blue), hold the power button down for 5 seconds until you hear a click, let go of the
+      power button and then press the button once to activate.
+
+#. Press enter on your putty window to start the flash process and wait. If the connection is successful you should see a large amount of scrolling text, this
+   is the script backing up the factory shipped firmware.
+#. Once the process is complete you can type ``curl http://10.42.42.42/flashURL?url=http://10.42.42.1/files/firmware.bin``
+#. The plug will restart and if everything is working correctly after a few seconds you should be able to press the button triggering the relay and turning the
+   blue led on.
+
+3. Smart Plug Configurations
+----------------------------
+
+Thanks to the existing work by `@ideasman69 <https://github.com/ct-Open-Source/tuya-convert/issues/66>`__ and
+`afalzon <https://github.com/arendst/Sonoff-Tasmota/wiki/Mirabella-Genio-Smart-Plug>`__
+which these adaptions created by `@cryptelli <https://community.home-assistant.io/u/cryptelli>`__ are based on.
+
+
+3.1 Brilliant Smart Plug
+************************
+
+.. code-block:: yaml
+
+    esphome:
+      name: brilliant_smart_plug
+      platform: ESP8266
+      board: esp01_1m
+
+    wifi:
+      ssid: 'WIFI'
+      password: 'WIFIPASS'
+
+    logger:
+
+    api:
+
+    ota:
+
+    output:
+      - platform: esp8266_pwm
+        id: power_led
+        pin:
+          number: GPIO12
+          inverted: true
+
+    light:
+      - platform: monochromatic
+        name: "Power LED"
+        output: power_led
+        id: red_led
+        default_transition_length: 0s
+
+    binary_sensor:
+      - platform: gpio
+        pin:
+          number: 14
+          mode: INPUT_PULLUP
+          inverted: true
+        name: "Power Button"
+        on_press:
+          - switch.toggle: relay
+      - platform: status
+        name: Status
+
+    switch:
+      - platform: gpio
+        name: "Brilliant Smart Plug"
+        pin: GPIO5
+        id: relay
+
+        # Turn off red LED to show blue when turned on
+        on_turn_on:
+          - light.turn_off:
```

Comment:
If 0s is the default transition length, we can simplify this: `light.turn_off: red_led` (same below)

Reply:
Addressed in latest commit, thanks for the suggestions.